### ISSUE TITLE
incorrect DATM namelist when user_datm.streams files are present

### ISSUE THREAD

**Issue by lvankampenhout** [ESMCI/cime#876]
Using cime5.2.0-alpha.7

## How to reproduce (Yellowstone)

Create a new case with DATM compset:
`/glade/u/home/lvank/cesm/cesm_tags2/clm4_5_13_r205/cime/scripts/create_newcase --case test_datm_nl -res f09_g16 --compset 1850_DATM%CRU_CLM50%BGC_SICE_SOCN_SROF_SGLC_SWAV --user-compset -mach yellowstone --pesfile /glade/u/home/lvank/cesm/cesm_tags2/clm4_5_13_r205/components/clm/cime_config/config_pes.xml`

Then generate the datm_in using ./case.setup and ./preview_namelists, it looks like this:

```
&datm_nml
  decomp = "1d"
  factorfn = "null"
  force_prognostic_true = .false.
  iradsw = 1
  presaero = .true.
  restfilm = "undefined"
  restfils = "undefined"
  wiso_datm = .false.
/
&shr_strdata_nml
  datamode = "CLMNCEP"
  domainfile = "/glade/p/cesmdata/cseg/inputdata/share/domains/domain.lnd.fv0.9x1.25_gx1v6.090309.nc"
  dtlimit = 5*1.5
  fillalgo = 5*"nn"
  fillmask = 5*"nomask"
  fillread = 5*"NOT_SET"
  fillwrite = 5*"NOT_SET"
  mapalgo = 5*"bilinear"
  mapmask = 5*"nomask"
  mapread = 5*"NOT_SET"
  mapwrite = 5*"NOT_SET"
  streams = "datm.streams.txt.CLMCRUNCEP.Solar 1 1901 1920",
      "datm.streams.txt.CLMCRUNCEP.Precip 1 1901 1920",
      "datm.streams.txt.CLMCRUNCEP.TPQW 1 1901 1920",
      "datm.streams.txt.presaero.clim_1850 1 1 1",
      "datm.streams.txt.topo.observed 1 1 1"
  taxmode = 5*"cycle"
  tintalgo = "coszen", "nearest", 2*"linear", "lower"
  vectors = "null"
/
```

Copy `user_datm.streams.txt.*` from somewhere, e.g. /glade/u/home/lvank/cesm/my_cases/test/test_datm_nl. Then ./preview_namelists gives the following datm_in:

```
&datm_nml
  decomp = "1d"
  factorfn = "null"
  force_prognostic_true = .false.
  iradsw = 1
  presaero = .true.
  restfilm = "undefined"
  restfils = "undefined"
  wiso_datm = .false.
/
&shr_strdata_nml
  datamode = "CLMNCEP"
  domainfile = "/glade/p/cesmdata/cseg/inputdata/share/domains/domain.lnd.fv0.9x1.25_gx1v6.090309.nc"
  dtlimit = 1.5
  fillalgo = "nn"
  fillmask = "nomask"
  fillread = "NOT_SET"
  fillwrite = "NOT_SET"
  mapalgo = "bilinear"
  mapmask = "nomask"
  mapread = "NOT_SET"
  mapwrite = "NOT_SET"
  streams =
  taxmode = "cycle"
  tintalgo = "linear"
  vectors = "null"
/
```

**Comment by billsacks** [comment]
@mvertens - is #916 (which was fixed by #915) a duplicate of this? If so, this can be closed.

**Comment by mvertens** [comment]
This was fixed by #915

**Comment by mvertens** [comment]
Yes - this was a duplicate. I'll close
